### PR TITLE
Remove Affinity

### LINKED ISSUE #3224: Remove AFFINITY
based on a Skype discussion with the original authors (@icimrak, @ijancigova).
The feature is no longer used.

I'll prepare a PR, after the school.

### PR DESCRIPTION
**Pull request by fweik**
Fixes #3224.

### CODE REVIEW

**Review by KaiSzuttor — APPROVED** [review]

File: src/core/forces_inline.hpp [hunk 1]
```diff
@@ -252,14 +251,6 @@ inline void add_non_bonded_pair_force(Particle &p1, Particle &p2,
   /* bond creation and breaking                  */
   /***********************************************/
```
Comment (jngrad) [inline src/core/forces_inline.hpp]: this comment is now obsolete

**Review by jngrad — APPROVED** [review]
LGTM

**Comment by KaiSzuttor** [comment]
skip ci is not the best idea... you will never get a green light from ci now.

**Comment by fweik** [comment]
Well that is a bug, no?

**Comment by fweik** [comment]
bors r=KaiSzuttor

**Comment by KaiSzuttor** [comment]
yes

**Comment by KaiSzuttor** [comment]
i'm not sure, but I think bors will fail to push since the green status of CI is required

**Comment by KaiSzuttor** [comment]
I don't want to invest time for repairing this bug since we will have a transition to github actions soon

**Pull request merged**